MAUI/DataForm/grouping.md

```diff
@@ -0,0 +1,308 @@
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.Name == "Name")
+            {
+                e.DataFormGroupItem.ColumnCount = 2;
+            }
+        }
+    }
+
+
+{% endhighlight %}
+{% endtabs %}
+
+In the following image, for the `Name` group, the grid layout is loaded and for the `Details` group, linear layout is loaded:
+
+
+## Setting different column count
+
+The different `ColumnCount` can also be set for each group.
+
+{% tabs %}
+{% highlight c# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.Name == "Name")
+            {
+                e.DataFormGroupItem.ColumnCount = 2;
+            }
+            else if (e.DataFormGroupItem.Name == "Details")
+            {
+                e.DataFormGroupItem.ColumnCount = 3;
+            }
+        }
+    }
+
+
+{% endhighlight %}
+{% endtabs %}
+
+## Loading group in collapsed state
+
+By default, the group will be loaded in expanded state. You can collapse the group by setting the [IsExpanded] property of [DataFormGroupItem] class to `false`.
+
+{% tabs %}
+{% highlight c# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.Name == "Name")
+            {
+                e.DataFormGroupItem.IsExpanded = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Restricting the group expanding and collapsing
+
+The group being expanded or collapsed can be restricted by setting the [AllowExpandCollapse] to `false` in the [DataFormGroupItem].
+
+{% tabs %}
+{% highlight c# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.Name == "Name")
+            {
+                e.DataFormGroupItem.AllowExpandCollapse = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing the group editors visibility
+
+The data form group visibility can be changed by using the [IsVisible] property of [DataFormGroupItem].
+
+{% tabs %}
+{% highlight c# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.Name == "Name")
+            {
+                e.DataFormGroupItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
```

Comment:
Add about HeaderBackground , Padding and HeaderTextStyle properties of GroupItem